### PR TITLE
Add support for proxy defaults CRD

### PR DESCRIPTION
**Pull request by thisisnotashwin**
* Companion to https://github.com/hashicorp/consul-k8s/pull/328
* Docker image: `https://github.com/hashicorp/consul-k8s/pull/328`
* Testing
  * I added a test for OSS proxy-defaults

• My plagiarism of Luke's write-ups continue.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -140,7 +140,7 @@ jobs:
               -kubeconfig="$primary_kubeconfig" \
               -secondary-kubeconfig="$secondary_kubeconfig" \
               -debug-directory="$TEST_RESULTS/debug" \
-              -consul-k8s-image=hashicorpdev/consul-k8s:crd-controller-base-latest
+              -consul-k8s-image=ashwinvenkatesh/consul-k8s@sha256:27c86621f276b3cf8f4ee1d85ae0a0d7a0d5436d76265dc225782a16b5a60a83
```
Comment (ashwin-venkatesh) [inline .circleci/config.yml:143]: This is purely for tests. Will revert before merging.

File: templates/crd-servicedefaults.yaml [hunk 2]
```diff
@@ -2,16 +2,15 @@ apiVersion: apiextensions.k8s.io/v1beta1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.2.5
+    controller-gen.kubebuilder.io/version: v0.3.0
```
Comment (ashwin-venkatesh) [inline templates/crd-servicedefaults.yaml:5]: The updated version of `controller-runtime` updated the version of `controller-gen` causing the changes in this file that are purely cosmetic.

File: templates/crd-serviceresolver.yaml [hunk 3]
```diff
@@ -2,16 +2,15 @@ apiVersion: apiextensions.k8s.io/v1beta1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.2.5
+    controller-gen.kubebuilder.io/version: v0.3.0
```
Comment (ashwin-venkatesh) [inline templates/crd-serviceresolver.yaml:5]: The updated version of `controller-runtime` updated the version of `controller-gen` causing the changes in this file that are purely cosmetic.

File: test/acceptance/tests/controller/controller_test.go [hunk 4]
```diff
@@ -79,6 +79,13 @@ func TestController(t *testing.T) {
 					svcResolverEntry, ok := entry.(*api.ServiceResolverConfigEntry)
 					require.True(r, ok, "could not cast to ServiceResolverConfigEntry")
 					require.Equal(r, "bar", svcResolverEntry.Redirect.Service)
+
```
Comment (lkysow) [inline test/acceptance/tests/controller/controller_test.go:82]: Need to add test to controller_ent_test

**Review by lkysow — COMMENTED** [review]
So simple! 😃

File: test/acceptance/tests/controller/controller_test.go (hunk 4 above)
Comment (ashwin-venkatesh) [reply test/acceptance/tests/controller/controller_test.go:82]: DONE 😄

File: test/acceptance/tests/controller/controller_namespaces_test.go [hunk 5]
```diff
@@ -134,6 +134,13 @@ func TestControllerNamespaces(t *testing.T) {
 					svcResolverEntry, ok := entry.(*api.ServiceResolverConfigEntry)
 					require.True(r, ok, "could not cast to ServiceResolverConfigEntry")
 					require.Equal(r, "bar", svcResolverEntry.Redirect.Service)
+
+					// proxy-defaults
+					entry, _, err = consulClient.ConfigEntries().Get(api.ProxyDefaults, "global", queryOpts)
```
Comment (lkysow) [inline test/acceptance/tests/controller/controller_namespaces_test.go:139]: So your `Get()` call will need to be in the default consul namespace. this means removing `queryOpts` and using `nil` (which will default to default ns) or constructing a new one and explicitly setting it to `default`

**Comment by ashwin-venkatesh** [comment]
if it passes acceptance, it is acceptable 😛

**Review by lkysow — APPROVED** [review]

**Review by ishustava — APPROVED** [review]
Looks good, given that the acceptance tests pass!

**Pull request merged**